### ISSUE TITLE
Outlier detection needs to log and handle grpc-status sent as trailers

### ISSUE THREAD

**Issue by wenbozhu** [envoyproxy/envoy#8228]
*Title*: Outlier detection needs to log and handle grpc-status sent as trailers

*Description*:
>Outlier detection currently doesn't check response trailers, which may contain an error grpc-status when a gRPC stream is closed. 

*Relevant Links*:
>See the [discussion](https://github.com/envoyproxy/envoy/issues/860#issuecomment-524453203) on the original issue for adding grpc-status support in Outlier Detection.

**Comment by cpakulski** [comment]
Sorry, I was away - will comment within a day or two. Thanks!

**Comment by rcash** [comment]
Hi @wenbozhu  and @cpakulski I'm a student in UT Austin's Virtualization & Containers class. Are you still looking for help in adding support for gRPC response trailers? Thanks!

**Comment by wenbozhu** [comment]
We probably need some design inputs from @cpakulski  first ...  

@htuch

**Comment by htuch** [comment]
Yeah, I think if you want to help drive the design here and propose a strawman @rcash that might be a reasonable start.

**Comment by cpakulski** [comment]
@rcash, @wenbozhu Please look at https://github.com/envoyproxy/envoy/pull/7942 as that solved similar case of passing gRPC codes to outlier detector. Let me know if you get stuck in design or understanding the code flow.

**Comment by luke-han** [comment]
Any updates?

**Comment by cpakulski** [comment]
@luke-han Can you describe your use case? What value in trailers do you want to use?

**Comment by luke-han** [comment]
@cpakulski 
I tried applying a circuit breaker to a gRPC-based LLM streaming API, but it didn't work as expected. 
Even when the same error code repeatedly occurred, the circuit breaker did not trigger. While investigating related issues, I found this issue.
After checking the logs, I noticed that the status is present in the trailer. 

Here is a portion of the logs for reference:
```
2025-03-05T05:11:23.819729Z debug   envoy http external/envoy/source/common/http/conn_manager_impl.cc:1878  [Tags: "ConnectionId":"1230","StreamId":"12171118623636036793"] encoding headers via codec (end_stream=false):
':status', '200'
'content-type', 'application/grpc'
...

2025-03-05T05:11:23.903098Z debug   envoy http external/envoy/source/common/http/conn_manager_impl.cc:1910  [Tags: "ConnectionId":"1230","StreamId":"12171118623636036793"] encoding trailers via codec:
'grpc-status', '14'
...
```
